### PR TITLE
Abstract TextEncoder as a polyfill

### PR DESCRIPTION
**Pull request by supl**
This PR abstracts the `TextEncoder` to a single utility. `TextEncoder` is polyfill to convert strings into Uint8Array(). I did this because it's used by the other file too.

### CODE REVIEW

File: request/builder.js [hunk 1]
```diff
@@ -610,15 +602,19 @@ class ContractExecutionRequestBuilder {
...
     const certVersion = new Uint8Array(view.buffer);
 
     const buffer = new Uint8Array(
-        contractIdEncoded.byteLength + contractArgument.byteLength +
-        certHolderId.byteLength + certVersion.byteLength);
+        contractIdEncoded.byteLength +
+        contractArgument.byteLength +
+        certHolderId.byteLength +
+        certVersion.byteLength,
+    );
```
Comment (supl) [inline request/builder.js:617]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 2]
```diff
@@ -610,15 +602,19 @@ class ContractExecutionRequestBuilder {
 
     const contractIdEncoded = new TextEncoder('utf-8').encode(this.contractId);
     const contractArgument = new TextEncoder('utf-8').encode(
-        this.contractArgument);
+        this.contractArgument,
+    );
```
Comment (supl) [inline request/builder.js:606]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 3]
```diff
@@ -382,8 +371,10 @@ class ContractsListingRequestBuilder {
     const contractIdEncoded = new TextEncoder('utf-8').encode(this.contractId);
 
     const buffer = new Uint8Array(
-        contractIdEncoded.byteLength + certHolderId.byteLength +
-        certVersion.byteLength);
+        contractIdEncoded.byteLength +
+        certHolderId.byteLength +
+        certVersion.byteLength,
+    );
```
Comment (supl) [inline request/builder.js:377]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 4]
```diff
@@ -279,19 +262,25 @@ class ContractRegistrationRequestBuilder {
...
-        contractId.byteLength + contractBinaryName.byteLength +
-        contractBytes.byteLength + contractProperties.byteLength +
-        certHolderId.byteLength + certVersion.byteLength);
+        contractId.byteLength +
+        contractBinaryName.byteLength +
+        contractBytes.byteLength +
+        contractProperties.byteLength +
+        certHolderId.byteLength +
+        certVersion.byteLength,
+    );
```
Comment (supl) [inline request/builder.js:283]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 5]
```diff
@@ -279,19 +262,25 @@ class ContractRegistrationRequestBuilder {
...
     const contractId = new TextEncoder('utf-8').encode(this.contractId);
     const contractBinaryName = new TextEncoder('utf-8').encode(
-        this.contractBinaryName);
+        this.contractBinaryName,
+    );
     const contractBytes = this.contractByteCode;
     const contractProperties = new TextEncoder('utf-8').encode(
-        this.contractProperties);
+        this.contractProperties,
+    );
```
Comment (supl) [inline request/builder.js:270]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 6]
```diff
@@ -279,19 +262,25 @@ class ContractRegistrationRequestBuilder {
 
     const contractId = new TextEncoder('utf-8').encode(this.contractId);
     const contractBinaryName = new TextEncoder('utf-8').encode(
-        this.contractBinaryName);
+        this.contractBinaryName,
+    );
```
Comment (supl) [inline request/builder.js:266]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 7]
```diff
@@ -33,14 +15,15 @@ class Validator {
...
 
-    if (typeof input === 'undefined' ||
-        input === null ||
-        input.constructor !== type ||
-        (input.constructor === Number && input < 0)
+    if (
+      typeof input === 'undefined' ||
+      input === null ||
+      input.constructor !== type ||
+      (input.constructor === Number && input < 0)
```
Comment (supl) [inline request/builder.js:26]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 8]
```diff
@@ -33,14 +15,15 @@ class Validator {
       return;
     }
 
-    if (type.name === 'Uint8Array' && (input instanceof Uint8Array)) {
+    if (type.name === 'Uint8Array' && input instanceof Uint8Array) {
```
Comment (supl) [inline request/builder.js:18]: This part is formatted automatically by eslint. Can ignore it.

File: request/builder.js [hunk 9]
```diff
@@ -1,22 +1,4 @@
...
-   * @param {string} string
-   * @return {Uint8Array}
-   */
-  encode(string) {
-    return !string ?
-        new Uint8Array() :
-        new Uint8Array(
-            jsrsasign.hextoArrayBuffer(jsrsasign.utf8tohex(string)));
-  }
-}
```
Comment (supl) [inline request/builder.js:19]: The main change is here.
This part is moved to the `polyfill/text_encoder.js` file

**Comment by supl** [comment]
This PR needs to be reviewed after #76 because the test framework is changed.

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**